granary/microformats2.py

```diff
@@ -356,23 +374,30 @@ def json_to_html(obj):
   prop = first_props(props)
   prop.setdefault('uid', '')
   author = prop.get('author')
-  if author:
-    author['type'].append('p-author')
 
   # if this post is itself a like or repost, link to its target(s). (do this and
   # rsvp below *before* content since they set props['name'] if necessary.)
   likes_and_reposts = []
-  for verb in 'like', 'repost':
-    if ('h-as-%s' % verb) in obj['type']:
-      if not props.get('name'):
-        props['name'] = ['%ss this.' % verb]
-      likes_and_reposts += ['<a class="u-%s u-%s-of" href="%s"></a>' %
-                            (verb, verb, url) for url in props.get(verb)]
+  if 'h-as-like' in obj['type']:
+    if not props.get('name'):
+      props['name'] = ['likes this.']
+    likes_and_reposts += ['<a class="u-like u-like-of" href="%s"></a>' %
```

Comment:
crud, I see your point. 

Line 190 is checking the length of the dict (to make sure there is only one property, "url"), not the length of the list of urls, so multiple urls should still be OK, but I still agree assuming likes will be strings is fragile. I'll reunify these two cases, and check for isinstance dict on both of them.